src/linear_algebra/smodeq.lean

```diff
@@ -14,66 +15,99 @@ import ring_theory.ideal.quotient
 open submodule
 open_locale polynomial
 
-variables {R : Type*} [ring R]
-variables {M : Type*} [add_comm_group M] [module R M] (U U₁ U₂ : submodule R M)
+variables {M X : Type*} [add_comm_group M] [set_like X M] [add_subgroup_class X M] (U U₁ U₂ : X)
+variables {R : Type*} [ring R] [module R M] (Ug : add_subgroup M) (Us : submodule R M)
 variables {x x₁ x₂ y y₁ y₂ z z₁ z₂ : M}
 variables {N : Type*} [add_comm_group N] [module R N] (V V₁ V₂ : submodule R N)
 
-/-- A predicate saying two elements of a module are equivalent modulo a submodule. -/
+/-- A predicate saying two elements of a additive group are equivalent modulo a subgroup. -/
 def smodeq (x y : M) : Prop :=
-(submodule.quotient.mk x : M ⧸ U) = submodule.quotient.mk y
```

Comment:
Another solution here would be to eliminate the distinction between `submodule.quotient` and `add_subgroup.quotient`; they're already defined in terms of each other.

Did you attempt that at all @Vierkantor?

Reply:
I think I tried that a while ago before the redefinition of the quotients, and it was quite a bit of straightforward work so not too bad if you have some time. We did get the `⧸` notation out of that attempt though.